solutions/0000-0099/0012-integer-to-roman-medium.md

```diff
@@ -0,0 +1,148 @@
+---
+description: >-
+  Author: @jessicaribeiroalves | https://leetcode.com/problems/integer-to-roman/
+---
+
+# 0012 - Integer to Roman (Medium)
+
+## Problem Link
+
+<https://leetcode.com/problems/integer-to-roman/>
+
+## Problem Statement
+
+Roman numerals are represented by seven different symbols: `I`, `V`, `X`, `L`, `C`, `D` and `M`.
+
+```Python
+Symbol       Value
+I             1
+V             5
+X             10
+L             50
+C             100
+D             500
+M             1000
+```
+
+For example, `2` is written as `II` in Roman numeral, just two ones added together. `12` is written as `XII`, which is simply `X + II`. The number `27` is written as `XXVII`, which is `XX + V + II`.
+
+Roman numerals are usually written largest to smallest from left to right. However, the numeral for four is not `IIII`. Instead, the number four is written as `IV`. Because the one is before the five we subtract it making four. The same principle applies to the number nine, which is written as `IX`. There are six instances where subtraction is used:
+
+- `I` can be placed before `V` (5) and `X` (10) to make 4 and 9.
+- `X` can be placed before `L` (50) and `C` (100) to make 40 and 90.
+- `C` can be placed before `D` (500) and `M` (1000) to make 400 and 900.
+
+Given an integer, convert it to a roman numeral.
+
+**Example 1:**
+
+```Python
+Input: num = 3
+Output: "III"
+Explanation: 3 is represented as 3 ones.
+```
+
+**Example 2:**
+
+```Python
+Input: num = 58
+Output: "LVIII"
+Explanation: L = 50, V = 5, III = 3.
+```
+
+**Example 3:**
+
+```Python
+Input: num = 1994
+Output: "MCMXCIV"
+Explanation: M = 1000, CM = 900, XC = 90 and IV = 4.
+```
+
+**Constraints:**
+
+- `1 <= num <= 3999`
+
+## Approach 1: Iterating over the dictionary entries
+
+The solution used was iterating over the entries of a dictionary and for each entry, while the remaining value is greater than or equal to the key value of the dictionary, append the corresponding character(s) to the `result` and subtract the key from the remaining value.
+
+For example, if we consider the given integer `num = 17`, after starting the iteration over the dictionary, the algorithm checks if the integer `17` is greater than or equal to the first integer value in the dictionary, which is `1000`. Since it's not, the code in the while loop is not executed and the next iteration starts checking if `17` is greater than or equal to `900` and so on until the iteration checks if `17` is greater than or equal to `10`. In this iteration the code in the while loop is executed so that the `result` variable is concatenated with the respective Roman numeral which is `'X'` and the remaining value becomes `7`. For the next iteration, `7` is not greater than or equal to `9` but on the next one, when it's greater than `5`, `result` becomes `'XV'` and the remaining value becomes `2`. So keeping that logic, after the final iteration the final result will be `'XVII'`.
```

Comment:
No backticks as it's stated

```
Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
```

Reply:
Thanks for the feedback @wingkwong 
I've replaced the back-ticks with LaTex for the variables and formulas (as stated) but not for the examples of values and roman numerals, because I thought they didn't fit the statement. Let me know if I got it wrong, and I'll update again if necessary.
Now it looks like this:
<img width="720" alt="Screen Shot 2022-10-19 at 10 52 52 AM" src="https://user-images.githubusercontent.com/42510740/196733426-40ce5843-e9ce-438c-b027-1c81743b2140.png">

Reply:
Only apply on what you made. Problem Statement is excluded.